### ISSUE TITLE
No match with chr1 between the reference index file and the input vcf file

### ISSUE THREAD

**Issue by jingydz** [stat-lab/MOPline#11]
![image](https://github.com/stat-lab/MOPline/assets/45711930/6e1d76ed-83b8-4de9-9808-01dbb92f1969)
I followed the structure of mopline in the figure.

Why I got the information: "No match with chr1 between the reference index file and the input vcf file", and my  "sample.Merge.ALL.vcf" file changed into empty.

**Comment by jingydz** [comment]
first step: mopline merge_7tools --sample samplename -rl 150
second step: mopline create_cov \
-b /xxx/samplename/samplename.bam \
-r /xxx/hg38.fa \
-rl 150 -n 4
third step: mopline add_cov --sample_list samplename.list --toolset 7tools --vcf_dname Merge_7tools -n 2

**Comment by jingydz** [comment]
less samplename/Manta/Manta.samplename.vcf
##ALT=<ID=INV,Description="Inversion">
##ALT=<ID=DEL,Description="Deletion">
##ALT=<ID=INS,Description="Insertion">
##ALT=<ID=DUP:TANDEM,Description="Tandem Duplication">

The header mentioned above is from the VCF result file of the sample processed by the Manta software. After running Manta, I also used the convertInversion.py script to convert some BNDs into INVs. Is this the correct input for mopline?

**Comment by stat-lab** [comment]
I suppose your data is human data. By default, the human reference build is GRCh37 (specify --build 38 if you use build 38). Is the chr name 1, 2, ... X in your vcf file ?

**Comment by jingydz** [comment]
Yes, there are chr1, 2, 3, ... in my vcf file.

Thank you. 

I run the command "mopline add_cov --sample_list samplename.list --toolset 7tools --build 38 --vcf_dname Merge_7tools -n 2". and it appears that my wham file has not been recognized.

**Comment by jingydz** [comment]
![image](https://github.com/stat-lab/MOPline/assets/45711930/610074d1-c587-4e80-b6ff-18ba4a611bc1)
Is the VCF file input for WHAM not containing genotypes?

**Comment by stat-lab** [comment]
Your reference genome seems GRCh38 or hg19. If hg19 is used, remove 'chr' prefix of the chromosome name in all your data, including vcf, fasta, and bam. If you use GRCh38 reference, add --build 38 option to your command.

**Comment by stat-lab** [comment]
Your Wham vcf contains genotype information.

To solve your previous error, an index file of your reference fasta used may be specified with --ref_index option.

**Comment by jingydz** [comment]
“Your Wham vcf contains genotype information.” The results from running the Wham software do not include genotypes, which I noticed is the same as in your example with my data, also lacking genotypes. By adding the --build 38 option, I have successfully run mopline add_cov and obtained the updated result file samplename.Merge.ALL.vcf.

**Comment by jingydz** [comment]
Hi, I have another question.
After running CNVnator, I got the cnv file and vcf file by cnvnator2VCF.pl.
For example, there are 7160 records in sample.cnv, and the same number of variant sites are also present in the VCF file.
But after using MOPline/scripts/run_SVcallers/convert_CNVnator_vcf.pl, there are only 2645 records in the cnvnator.sample.vcf.

During this conversion process, has too many variant sites been filtered out?

**Comment by jingydz** [comment]
Hi, Where can I find a detailed explanation of the following file, such as what the second column represents as positions with a step size of 50bp, and what do the other columns represent?

$ less sample/Cov/sample.chr1.cov.gz |head -n 3
chr1    10000   19.3    4.6     9       0
chr1    10050   39.2    13.7    1       0
chr1    10100   21.2    12.5    1       30

**Comment by jingydz** [comment]
Thank you so much for your quickly response!

I have another question. When I executed create_cov, I have a question about the runtime in your table. Why does the runtime increase when the number of cores increases?
![image](https://github.com/stat-lab/MOPline/assets/45711930/604f032d-919d-4e32-9ffd-dc10c192253f)

When I run it using my data, I find the result is different with yours.
My bam file size is 44G. 
When I used one core, it cost 1h50min for the first time, and it used 1h42min for the second time.
When I used two cores, it cost 54min.
When I used three cores, it cost 41min.
When I used four cores, it cost 33min.
When I used five cores, it cost 29min.

This is just the time required for running a single sample. When I use mopline's create_cov for multiple samples simultaneously, the time for each sample increases exponentially. Can you help me explain why this is happening?

Additionally, I noticed that in your README, the parameter for jointcall seems to be incorrect? It should be --md rather than --id. 
![image](https://github.com/stat-lab/MOPline/assets/45711930/bf203293-4c46-49c1-b21c-dfffb07bd345)

**Comment by stat-lab** [comment]
The create_cov with multi-threads treats the genome per chromosome. So, Specifying a thread number greater than the chromosome number is invalid. For multiple samples, I recommend you to run multiple jobs with subsets of samples but not to run a job sequentially for all samples.

The -id option for joint call is incorrect, but this has been fixed in the latest README on the GitHub.

**Comment by jingydz** [comment]
So, when I run add_cov, can I also split the entire sample_list into several smaller subsets of sample_list? 

For example: 
mopline add_cov --sample_list sample.list.1 --toolset 7tools --build 38 --vcf_dname Merge_7tools -n 4; 
mopline add_cov --sample_list sample.list.2 --toolset 7tools --build 38 --vcf_dname Merge_7tools -n 4; 
mopline add_cov --sample_list sample.list.3 --toolset 7tools --build 38 --vcf_dname Merge_7tools -n 4... and so on.

**Comment by stat-lab** [comment]
Yes. If you have a job management system such as LSF and Slum, you can use the scripts, run_batch_LSF.pl, run_batch_slurm.pl, and so on. If not, you can process the jobs in batches (e.g., nohub mopline add_cov --sample_list sample.list.1 --toolset 7tools --build 38 --vcf_dname Merge_7tools -n 4 &).
